comictaggerlib/settingswindow.py

```diff
@@ -235,6 +234,100 @@ def disconnect_signals(self) -> None:
         self.twLiteralReplacements.cellChanged.disconnect()
         self.twValueReplacements.cellChanged.disconnect()
 
+        self.sources: dict = {}
+        self.generate_source_option_tabs()
+
+    def generate_source_option_tabs(self) -> None:
```

Comment:
I want this function in it's own file and pass in as an argument the widget that everything will be built into, the config, and the talkers